src/components/dialogs/filter/criteria-based/filter-properties.tsx

```diff
@@ -100,74 +96,47 @@ export const filterPropertiesYupSchema = {
                 return areArrayElementsUnique(names);
             }
         ),
+    [FREE_FILTER_PROPERTIES]: yup
+        .array()
+        .of(
+            yup.object().shape({
+                [PROPERTY_NAME]: yup.string().required(),
+                [PROPERTY_VALUES]: yup
+                    .array()
+                    .of(yup.string())
+                    .test(
+                        'can not be empty if not line',
+                        'YupRequired',
+                        (values, context) =>
+                            propertyValuesTest(values, context, false)
+                    ),
+            })
+        )
+        .test(
+            'distinct names',
+            'filterPropertiesNameUniquenessError',
+            (properties, context) => {
+                // with context.from[length - 1], we can access to the root fields of the form
+                const rootLevelForm = context.from![context.from!.length - 1];
+                const filterType = rootLevelForm.value[FILTER_TYPE];
+                if (filterType !== FilterType.CRITERIA_BASED.id) {
+                    // we don't test if we are not in a criteria based form
+                    return true;
+                }
+                const names = properties! // never null / undefined
+                    .filter((prop) => !!prop[PROPERTY_NAME])
+                    .map((prop) => prop[PROPERTY_NAME]);
+                return areArrayElementsUnique(names);
+            }
+        ),
 };
 
-function fetchPredefinedProperties() {
```

Comment:
this function is called for each child component, in my opinion it will be better to let it here so we do the fetch once and we pass the predefined properties as a prop to the free properties sub-components

Reply:
Agreed, I've mutualized it in a state in FilterProperties